engine/execution/computation/computer/result_collector.go

```diff
@@ -290,13 +298,29 @@ func (collector *resultCollector) Finalize(
 		return nil, fmt.Errorf("failed to provide execution data: %w", err)
 	}
 
-	collector.result.ExecutionDataID = executionDataID
-
-	collector.result.ExecutionResult = flow.NewExecutionResult(
+	executionResult := flow.NewExecutionResult(
 		collector.parentBlockExecutionResultID,
 		collector.result.ExecutableBlock.ID(),
-		collector.result.Chunks,
+		collector.chunks,
 		collector.result.ConvertedServiceEvents,
-		collector.result.ExecutionDataID)
+		executionDataID)
+
+	collector.result.ExecutionReceipt = &flow.ExecutionReceipt{
+		ExecutionResult:   *executionResult,
+		Spocks:            collector.spockSignatures,
+		ExecutorSignature: crypto.Signature{},
+		ExecutorID:        collector.signer.NodeID(),
+	}
+
+	receiptId := collector.result.ExecutionReceipt.ID()
+	receiptSignature, err := collector.signer.Sign(
+		receiptId[:],
+		collector.receiptHasher)
+	if err != nil {
+		return nil, fmt.Errorf("could not sign execution result: %w", err)
+	}
+
+	collector.result.ExecutionReceipt.ExecutorSignature = receiptSignature
```

Comment:
```suggestion
	// unsigned receipt
	receipt := &flow.ExecutionReceipt{
		ExecutionResult:   *executionResult,
		Spocks:            collector.spockSignatures,
		ExecutorSignature: crypto.Signature{},
		ExecutorID:        collector.signer.NodeID(),
	}

	receiptId := receipt.ID()
	receiptSignature, err := collector.signer.Sign(
		receiptId[:],
		collector.receiptHasher)
	if err != nil {
		return nil, fmt.Errorf("could not sign execution result: %w", err)
	}
	
	// receipt is signed
	receipt.ExecutorSignature = receiptSignature
	
	collector.result.ExecutionReceipt = receipt
```

Reply:
Even though we removed `GenerateExecutionReceipt`, and still think it's worth to extract this logic into a function, so that it's clear here, for instance:
```
receipt := GenerateAndSignReceipt(unsignedReceipt, collector.signer)
```

Reply:
good point.  well do